src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java

```diff
@@ -102,10 +97,10 @@ public void setMax(int max) {
             TokenTypes.LITERAL_IF,
             TokenTypes.LITERAL_ELSE,
             TokenTypes.LITERAL_SWITCH,
-            TokenTypes.LITERAL_CASE,
-            TokenTypes.LITERAL_TRY,
-            TokenTypes.LITERAL_CATCH,
```

Comment:
lets, not mix problems. if try-catch does not work, I now try to remember some private discussions with original contributor about this, I will try to find what is left. 
But still, lets do try-catch separately, separate issue.